gnucash/gnome/dialog-fincalc.c

```diff
@@ -153,12 +157,12 @@ fi_to_gui(FinCalcDialog *fcd)
 
     pmt = double_to_gnc_numeric (fcd->financial_info.pmt, 100000, GNC_HOW_RND_ROUND_HALF_UP);
 
-    commodity = gnc_default_currency ();
+    precision = gtk_spin_button_get_value_as_int(GTK_SPIN_BUTTON(fcd->precision));
+    scu = (int64_t) pow(10, precision);
```

Comment:
Copy the [lookup table from gnc-numeric.cpp](https://github.com/Gnucash/gnucash/blob/maint/libgnucash/engine/gnc-numeric.cpp#L50). Much faster and no floating point to screw things up. But better yet, see below.